nixos/modules/programs/steam.nix

```diff
@@ -227,6 +238,8 @@ in {
         allowedTCPPorts = [ 27040 ]; # Data transfers
       })
     ];
+
+    boot.kernel.sysctl."vm.max_map_count" = lib.mkIf cfg.increaseMMapLimit (lib.mkDefault 2147483642);
```

Comment:
Maybe just pump it directly in https://github.com/NixOS/nixpkgs/blob/ebf6177371f8fdc8ff466e4ab6c90fe1347db0db/nixos/modules/config/sysctl.nix#L83 if needed, it's already 1048576.